python/sbp/acquisition.py

```diff
@@ -295,9 +333,16 @@ def from_binary(self, d):
     the message.
 
     """
-    p = MsgAcqResultDepB._parser.parse(d)
-    for n in self.__class__.__slots__:
-      setattr(self, n, getattr(p, n))
+    try:
+      self._from_binary(d)
+    except:
+      print traceback.print_exc()
+
+  def __getitem__(self, item):
+    return getattr(self, item)
+
+  def _get_embedded_type(self, t):
+    return globals()[t]
```

Comment:
Right away I can't see simple way to update that mapping automatically when new message is added to SBP specifications. globals() is simple and not even so frowned upon as eval(). Unless someone can point out a security hole created by this I'd leave it like this for now. I'll spent a few more moments thinking if something more elegant and as simple comes up though.